src/App.tsx

```diff
@@ -1,26 +1,157 @@
+  const [error, setError] = useState<ErrorType | null>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState<FilterType>(FilterType.All);
+  const [tmpTodo, setTmpTodo] = useState<Omit<Todo, 'userId'> | null>(null);
+  const [updatedTodosId, setUpdatedTodosId] = useState<number[]>([]);
+
+  const downloadTodos = async () => {
+    try {
+      const tmp = await getTodos();
+
+      setTodos(tmp);
+      setFilteredTodos(tmp);
+    } catch {
+      setError(ErrorType.UnableToLoad);
+    }
+  };
+
+  useEffect(() => {
+    downloadTodos();
+    inputRef.current?.focus();
+  }, []);
+
+  useEffect(() => {
+    if (todos) {
+      switch (query) {
+        case FilterType.Active:
+          setFilteredTodos(todos.filter(todo => !todo.completed));
+          break;
+        case FilterType.Completed:
+          setFilteredTodos(todos.filter(todo => todo.completed));
+          break;
+        default:
+          setFilteredTodos(todos);
+          break;
+      }
+    }
+  }, [todos, query]);
+
+  const returnLeftNumber = () => {
+    return todos.filter(todo => todo.completed != true).length;
+  };
+
+  const handleSetQuery = (passedQuery: FilterType) => {
+    setQuery(passedQuery);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (inputRef.current?.value.trim()) {
+      inputRef.current.disabled = true;
+      setTmpTodo({
+        title: inputRef.current.value.trim(),
+        id: -1,
+        completed: false,
+      });
+
+      postTodo(inputRef.current.value.trim())
+        .then(response => {
+          setTodos(prev => [...prev, response]);
+
+          if (inputRef.current) {
+            inputRef.current.value = '';
+          }
+        })
+        .catch(() => setError(ErrorType.UnableToAdd))
+        .finally(() => {
+          if (inputRef.current) {
+            inputRef.current.disabled = false;
+            inputRef.current.focus();
+          }
+
+          setTmpTodo(null);
+        });
+    } else {
+      setError(ErrorType.EmptyTitle);
+    }
+  };
+
+  const handleRemoveTodoById = (id: number) => {
+    setUpdatedTodosId(prev => [...prev, id]);
+
+    removeTodo(id)
+      .then(() => {
+        setTodos(prev => prev.filter(el => el.id != id));
+      })
+      .catch(() => setError(ErrorType.UnableToDelete))
+      .finally(() => {
+        setUpdatedTodosId(prev => prev.filter(el => el != id));
+        inputRef.current?.focus();
+      });
+  };
+
+  const handleRemoveAllComplited = () => {
+    const todosToRemoveIds = todos
+      .filter(todo => todo.completed)
+      .map(todo => todo.id);
+
+    setUpdatedTodosId(todosToRemoveIds);
+
+    todosToRemoveIds.map(id => {
+      handleRemoveTodoById(id);
+    });
+  };
+
+  const isClearAllCompletedActive = () => {
+    return !todos.every(todo => todo.completed);
+  };
```

Comment:
I know that it return now false only when all todos are completed, because I typed that in coment to your friend who said that button should be active only when all todos are completed couse the code comment saying that 
![image](https://github.com/user-attachments/assets/634b7c49-d977-448d-b2f6-7f679ebf8851)

Reply:
we are taking about two different buttons 😅 
toggle-all:
![image](https://github.com/user-attachments/assets/f0a446f1-f919-4ceb-9a8a-cf1b1a97ac9e)
and clear all:
![image](https://github.com/user-attachments/assets/9b9827b3-161b-4f96-a01d-04fe1a03136d)